Description: This patch applies upstream changes to fix a segfault bug when the first URI in try_files is shorter than the request URI.  The patch was derived from upstream.
Origin:      upstream, http://trac.nginx.org/nginx/changeset/4601/nginx
Bug:         http://trac.nginx.org/nginx/ticket/152
Bug-Ubuntu:  http://bugs.launchpad.net/bugs/1065555
--- a/src/http/ngx_http_core_module.c
+++ b/src/http/ngx_http_core_module.c
@@ -1240,7 +1240,7 @@
             reserve = len > r->uri.len - alias ? len - (r->uri.len - alias) : 0;
         }
 
-        if (reserve > allocated) {
+        if (reserve > allocated || !allocated) {
 
             /* 16 bytes are preallocation */
             allocated = reserve + 16;
